### DISCUSSION TITLE
nova Impersonation

### DISCUSSION THREAD

**Discussion by nikoo-eshratabadi** [laravel/nova-issues#6407] [Q&A] [answered]
Hi everyone,
I need to use the built-in nova impersonation feature. is there any way i can add an icon and show it inline?

**Comment by webard** [comment]
When you add Trait `Impersonatable` to `User` model and define `canImpersonate` and `canBeImpersonated` methods, Impersonate icon shows automatically on User resource.

**Reply by nikoo-eshratabadi** [reply]
No it doesn't for me! It stays in the drop down menu -> ...

**Reply by webard** [reply]
If you want icons directly in resource or on table view, use this extension: https://novapackages.com/packages/digital-creative/icon-action-toolbar.

Works perfectly.

**Reply by nikoo-eshratabadi** [reply, marked as answer]
thanks for replying @webard 
i've seen this before actually but i don't know how to add it to a built-in function. looks to me this extension can be added to an action only
